### ISSUE TITLE
Throw an error for -> with an empty right-hand side

### ISSUE THREAD

**Issue by KronicDeth** [KronicDeth/intellij-elixir#218]
Port https://github.com/elixir-lang/elixir/commit/e4db37c1b0c030085b2a03f991fb103590eedbf7

**Comment by sh41** [comment]
Coming back to this one after a long time, and the interesting part turned out to be upstream rather than here.

The Elixir commit this issue links to is real, but it did not last. `throw_empty_stab_clause` was introduced on 2015-11-11 and replaced by `warn_empty_stab_clause` on 2015-11-16 — five days later. The grammar rule went from throwing to yielding a `nil` right-hand side and emitting a warning instead. That is still how it works: on Elixir 1.15 and on current `main`, `fn -> end` compiles, with

```
warning: an expression is always required on the right side of ->. Please provide a value after ->
```

Checked on Elixir 1.15.0/OTP 26 — it warns, and exits 0.

So this issue was filed on 2015-12-05, about nineteen days after the error it asks for had already been taken back out of Elixir. That is nobody's fault; the link pointed at a commit on the `v1.14` line, and Elixir's history was later rewritten, which makes the change hard to follow past 1.14.5 unless you go looking for it under its new hash.

The consequence is that the request as titled should not be built. If the plugin raised an error for `fn -> end`, it would be rejecting a program `elixirc` accepts — the editor would be stricter than the compiler, which is the wrong way round. The plugin currently parses `fn -> end` without an error, and there is a test pinning exactly that, so what is here today already agrees with Elixir.

What does survive is the smaller half: **Elixir warns, and the plugin says nothing at all.** An empty `->` right-hand side is almost always a mistake, and you presently find out at compile time rather than while typing. Matching the compiler's warning in the editor is a reasonable thing to want, and it is a much smaller change than the original ask — an inspection, rather than anything touching the grammar.

I'm leaving this open for that, rather than closing it. Read it from here on as "warn on an empty `->` right-hand side, as Elixir does" — not "throw an error", which the title still says and which is no longer the right behaviour.

This issue was looked at as part of an AI-assisted review of the backlog, and a human checked this reply before it was posted. If any of it is wrong, please say so here.
